admin_solutions/authentication.adoc

```diff
@@ -0,0 +1,166 @@
+$ openshift start master --public-master=https://myapiserver.com:8443 --write-config=githubconfig
+----
++
+[NOTE]
+====
+If you are installing by Ansible, then you must provide Ansible with the
+identityProvider configuration. When installing by Ansible, you probably do not
+want to be manually modifying your configuration, because you would lose the
+modifications when you re-ran your install, or upgraded. This topic shows you
+how to manually handle your configuration, but Ansible users should see ????.
+
+NeedInfo: there is a place you can plug in the identity provider config and
+Ansible will drop it into your config for you (see Jason or Scott for details).
+====
++
+[NOTE]
+====
+Using `openshift start master` on its own would auto-detect hostnames, but
+GitHub must be able to redirect to the exact host name that you specified when
+registering the application. For this reason, you cannot auto-detect the ID
+because it might redirect to the wrong address. Instead, you must specify the
+hostname that web browsers use to interact with your OpenShift cluster.
+====
+. Edit the new *_master-config.yaml_* file's `*identityProviders*` stanza.
+. Copy
+link:../install_config/configuring_authentication.html#GitHub[the example `*GitHubIdentityProvider*` configuration]
+and paste it to replace the existing stanza.
+. Make the following modifications to the `*identityProviders*` stanza:
+.. Change the provider `*name*` to match the callback URL you configured on
+GitHub.
++
+For example, if you defined the callback URL as
+`https://myapiserver.com:8443/oauth2callback/github/` then the `*name*` must be
+`github`.
+.. Change the `*clientID*` to the Client ID from GitHub that you
+link:../admin_solutions/authentication.html#register-app-on-github[registered previously].
+.. Change the `*clientSecret*` to the Client Secret from GitHub that you
+link:../admin_solutions/authentication.html#register-app-on-github[registered previously].
+.. Change the `*organizations*` to include a list of one or more GitHub
+organizations to which a user must have membership in order to authenticate. If
+specified, only GitHub users that are members of at least one of the listed
+organizations will be allowed to log in. If this is not specified, then any
+person with a valid GitHub account can log in.
+. Save your changes and close the file.
+. Start the OpenShift API server, specifying the configuration file you just
+modified:
++
+----
+$ openshift start master --config=<path/to/modified/config>/master-config.yaml
+----
+
+Once configured, any user logging in to the OpenShift web console will be
+prompted to log in using their GitHub credentials. On their first login, the
+user must click *authorize application* to permit GitHub to use their username,
+password, and organization membership with OpenShift. The user is the redirected
+back to the web console.
+
+=== Creating Users
+
+A common question with people who are new to OpenShift is "How can I create
+users?"
```

Comment:
The above might be a little to simple as a topic introduction. What's below should work fine as the intro. Although, we're creating users specifically with Google OAuth, so maybe a title change to "Creating Users with Google OAuth" might be better to differentiate between any other topics using different methods.